docs/data/material/migration/migrating-to-v6/migrating-to-pigment-css.md

```diff
@@ -0,0 +1,542 @@
+ */
+const pigmentConfig = {}; // we will refer to this later
+
+module.exports = withPigment(nextConfig, pigmentConfig);
+```
+
+</codeblock>
+
+Finally, import the stylesheet at the top of the layout file.
+
+```diff title="app/layout.(js|tsx)"
+ import type { Metadata } from 'next';
+ import { Inter } from 'next/font/google';
+
++ import '@mui/material-pigment-css/styles.css';
+
+ export default function RootLayout(props) {
+   return (
+     <html lang="en">
+       <body className={`${inter.className}`}>
+         {props.children}
+       </body>
+     </html>
+   );
+ }
+```
+
+### Vite
+
+Install the Vite plugin as a dev dependency:
+
+<codeblock storageKey="package-manager">
+
+```bash npm
+npm install --save-dev @pigment-css/vite-plugin
+```
+
+```bash yarn
+yarn add -D @pigment-css/vite-plugin
+```
+
+```bash pnpm
+pnpm add -D @pigment-css/vite-plugin
+```
+
+</codeblock>
+
+Next, open vite config file, usually named `vite.config.js`, and add the plugin:
+
+```js
+import { defineConfig } from 'vite';
+import { pigment } from '@pigment-css/vite-plugin';
+
+const pigmentConfig = {}; // we will refer to this later
+
+export default defineConfig({
+  plugins: [
+    pigment(pigmentConfig),
+    // ... Your other plugins.
+  ],
+});
+```
+
+Finally, add the Pigment CSS stylesheet to the top of the main file.
+
+```diff title="src/main.(js|tsx)"
+ import * as React from 'react';
++ import '@mui/material-pigment-css/styles.css';
+ import App from './App';
+
+ ReactDOM.createRoot(document.getElementById('root')).render(
+   <React.StrictMode>
+     <App />
+   </React.StrictMode>,
+ );
+```
+
+## Configuring the theme
+
+Integrating Pigment CSS with Material UI requires you to configure the theme to the plugin.
+Add the following code to your [Next.js](#nextjs) or [Vite](#vite) config file:
+
+```diff
++ import { extendTheme, stringifyTheme } from '@mui/material';
+
++ const pigmentConfig = {
++   theme: extendTheme(),
++ };
+```
+
+If you don't have a custom theme, you are ready to go. Start a development server by running:
+
+<codeblock storageKey="package-manager">
+
+```bash npm
+npm run dev
+```
+
+```bash yarn
+yarn dev
+```
+
+```bash pnpm
+pnpm dev
+```
+
+</codeblock>
+
+Open the browser and navigate to the localhost URL, you should see the app running with Pigment CSS.
+
+## How it works
+
+When a Pigment CSS plugin is configured through a framework bundler, it intercepts the styling APIs that Material UI uses and replaces them with those from Pigment CSS instead. Pigment CSS then extracts the styles at build time and injects them into the stylesheet.
+
+If you come from Material UI v5, using Pigment CSS will be a paradigm shift in terms of writing styles.
+Since Pigment CSS is a build-time extraction tool, it does not support dynamic styles that depend on runtime variables. For example, Pigment CSS will throw an error for styles that depend on a state like the one below:
```

Comment:
```suggestion
Since Pigment CSS is a build-time extraction tool, it does not support dynamic styles that depend on runtime variables.
For example, Pigment CSS will throw an error for styles that depend on a state like the one below:
```